File: src/pynote/infrastructure/migrations/v0007_vertical_split_reset.py
from __future__ import annotations

import sqlite3


def migrate(connection: sqlite3.Connection, applied_at_us: int) -> None:
    """가로 분할 시절에 저장된 편집 분할 크기를 비운다.

    편집기가 입력기와 같은 슬롯을 쓰면서 분할이 세로로 바뀌었고,
    editor_split_left/right 는 이제 좌우가 아니라 상단 슬롯·하단 목록의 크기를
    담는다(명명 부채 — 열 이름은 repository·상태 계층 수정을 피하려고 유지한다).
    기존 값은 (목록, 편집기) 순서라 그대로 읽으면 편집기가 작아지는 방향으로
    의미가 뒤집히므로 NULL 로 되돌려 기본 비율을 다시 적용받게 한다.
    """
    connection.execute(
        """
        UPDATE document_ui_states
        SET editor_split_left = NULL, editor_split_right = NULL
        """
    )
    connection.execute(
        """
        UPDATE schema_version
        SET version = 7, applied_at_us = ?
        WHERE id = 1
        """,
        (applied_at_us,),
    )

File: NoteEx/core/src/storage/migrations/v0007_vertical_split_reset.cpp
#include "pynote/core/storage/migrations/v0007_vertical_split_reset.h"

namespace pynote::core::storage::migrations::v0007
{
	namespace
	{
		// 파이썬 원본 v0007_vertical_split_reset.py 의 migrate 가 발행하는 문장이다. 앞뒤 개행과 들여쓰기
		// 공백까지 원본 그대로이고 선언 순서가 곧 실행 순서다 - 정적 게이트가 이 리터럴을
		// 순서대로 뽑아 원본과 바이트 비교하므로 구분자는 SQL 로 고정한다. narrow 리터럴은
		// 이 기계에서 CP949 로 변환되므로 SQLite 에 UTF-8 을 넘기려면 u8 리터럴이어야 한다.
		constexpr const char8_t* SQL_RESET_SPLIT = u8R"SQL(
        UPDATE document_ui_states
        SET editor_split_left = NULL, editor_split_right = NULL
        )SQL";

		constexpr const char8_t* SQL_UPDATE_VERSION = u8R"SQL(
        UPDATE schema_version
        SET version = 7, applied_at_us = ?
        WHERE id = 1
        )SQL";
	}

	bool Migrate(C_DATABASE& _database, std::int64_t _nAppliedAtUs)
	{
		if (!_database.Execute(reinterpret_cast<const char*>(SQL_RESET_SPLIT))) { return(false); }

		return(ExecuteBoundInt64(_database, reinterpret_cast<const char*>(SQL_UPDATE_VERSION), _nAppliedAtUs));
	}
}
